Port plugin to sbt {1.0.x, 0.13.16}

suhasgaddam opened the pull request:
This requires coordination with https://github.com/47deg/sbt-org-policies/pull/510 and https://github.com/47deg/sbt-dependencies/pull/30.

suhasgaddam commented on project/ProjectPlugin.scala:
```diff
@@ -20,28 +21,48 @@ object ProjectPlugin extends AutoPlugin {
+                "org.tpolecat" % "tut-plugin" % "0.6.0",
+                sbtBinaryVersionValue,
+                scalaBinaryVersionValue),
+              "com.sksamuel.scrimage" %% "scrimage-core" % "2.1.8"
```
https://github.com/sksamuel/scrimage/issues/140

juanpedromoreno commented on src/main/scala/microsites/MicrositeKeys.scala:
```diff
@@ -256,6 +256,8 @@ trait MicrositeAutoImportSettings extends MicrositeKeys {
         .value
     },
     publishMicrosite := Def.task {
+      // I don't know how to fix this for sbt 1.0.
```
Maybe you can use as we did here:

https://github.com/47deg/sbt-org-policies/pull/510/files#diff-09ec74f5913120bc52ff45882b201224R88

fedefernandez commented on project/ProjectPlugin.scala:
```diff
@@ -20,28 +21,48 @@ object ProjectPlugin extends AutoPlugin {
+      addSbtPlugin("com.typesafe.sbt" % "sbt-ghpages" % "0.6.2"),
+      addSbtPlugin("com.typesafe.sbt" % "sbt-site"    % "1.3.0"),
       libraryDependencies ++= Seq(
+        "com.47deg"             %% "org-policies-core" % "0.6.1-SNAPSHOT",
```
This is `0.6.1` and `sbt-dependencies` is `0.6.1`. This is because the plan is to release in this order?
* sbt-org-policies: 0.6.0
* sbt-dependencies using 0.6.0
* sbt-org-policies: 0.6.1
* sbt-microsites using 0.6.1

suhasgaddam replied:
I was originally using a 0.6.1-SNAPSHOT locally to test these changes.
I was deciding releasing sbt-org-policies: 0.6.0 with a snapshot version of sbt-dependencies or releasing versions with no snapshot dependencies.

I like your way better. There is probably going to be at least one more version bump for all three sbt plugins after sbt-microsites gets published.

suhasgaddam commented:
@juanpedromoreno Can you re-review this PR?
I added `sbt-org-policies` and scalafmt changed some files...

I am specifically curious about https://github.com/47deg/sbt-microsites/pull/221/files#diff-4cfcd146802973e03a3a84659a33b5b0R275

`ghpagesPushSite.value` in a `Task` vs 

```scala
val ref = extracted.get(thisProjectRef)		
extracted.runAggregated[Unit](ghpagesPushSite in Global in ref, makeState)
```
in  a `Command`

`Task` -> sbt 1.0 change
`Command` -> sbt 0.13

xerial commented:
@juanpedromoreno When do you release this sbt-1.0 support? I'm really looking forward to the release!

juanpedromoreno commented:
Hi @xerial, we are working on it, in order to release it as soon as possible. At this moment we're having issues with `sbt-scripted`. Potentially, they are related to this one https://github.com/sbt/sbt/issues/3514

xerial commented:
@juanpedromoreno Thanks for preparing 0.7.0!

juanpedromoreno commented:
@xerial You're welcome, this was possible thanks to @suhasgaddam, who did the main work :)
